source/uwp/Renderer/lib/AdaptiveActionInvoker.cpp

```diff
@@ -15,14 +15,18 @@ namespace AdaptiveNamespace
 
     HRESULT AdaptiveActionInvoker::RuntimeClassInitialize(RenderedAdaptiveCard* renderResult) noexcept try
     {
-        m_renderResult = renderResult;
-        return S_OK;
-    }
-    CATCH_RETURN;
+        ComPtr<IRenderedAdaptiveCard> strongRenderResult = renderResult;
+        return strongRenderResult.AsWeak(&m_weakRenderResult);
+    } CATCH_RETURN;
 
     _Use_decl_annotations_ HRESULT AdaptiveActionInvoker::SendActionEvent(IAdaptiveActionElement* actionElement)
     {
-        return m_renderResult->SendActionEvent(actionElement);
+        ComPtr<IRenderedAdaptiveCard> strongRenderResult;
+        RETURN_IF_FAILED(m_weakRenderResult.As(&strongRenderResult));
+        RETURN_IF_FAILED(strongRenderResult == nullptr ? E_FAIL : S_OK);
```

Comment:
>RETURN_IF_FAILED(strongRenderResult == nullptr ? E_FAIL : S_OK); [](start = 8, length = 64)

Is this really an error? It's an expected case.

What CX and C# in their event source implementations is special-case when the delegate returns RPC_E_DISCONNECTED (I think that's the right error code).  In that case, the event source doesn't blow up, but does remove the listener from its list.